apps/web/src/hooks/useEffectOnce.ts

```diff
@@ -0,0 +1,14 @@
+import { useEffect, useRef } from 'react';
+
+export const useEffectOnce = (effect: () => void, condition: boolean) => {
+  const didRunRef = useRef(false);
+  const effectRef = useRef(effect);
+  effectRef.current = effect;
+
+  useEffect(() => {
+    if (condition && !didRunRef.current) {
+      didRunRef.current = true;
+      effectRef.current();
+    }
+  }, [condition]);
+};
```

Comment:
the hook that allows running the code only once when the condition is true

Reply:
I think I did something similar somewhere in the past, not sure if it still exists in the project 😄

Reply:
I couldn't find it... we should move such shared code to the hooks directory ;) but I guess it's not exist anymore

Reply:
i believe so it was a workaround so I hope we removed it 🫣